packages/dd-trace/src/encode/tags-processors.js

```diff
@@ -38,11 +38,12 @@ function truncateToLength (value, maxLength) {
   return value
 }
 
-function truncateSpan (span) {
+// normally the agent truncates the resource and parses it in certain scenarios (e.g. SQL Queries)
+function truncateSpan (span, truncateResource) {
```

Comment:
could we do something like 
```suggestion
function truncateSpan (span, shouldTruncateResourceName = true) {
```
with the default value we could also get rid of passing `true` for `packages/dd-trace/src/encode/agentless-ci-visibility.js`

Reply:
That was actually my first approach